### ISSUE TITLE
robots.txt

### ISSUE THREAD

**Issue by cloudrac3r** [cloudrac3r/bibliogram#11]
Currently there is no robots.txt file, which means search engines can visit anything on the site. This means search engines will crawl through whatever users and posts they can find, putting extra load on the site to index content that is unlikely to be found by people actually searching the web.

robots.txt would of course be editable by the owner of the instance, but what do you think would be sane defaults to start off with? Remember, [robots.txt supports excluding specific user-agents (or all of them) from specific paths.](https://www.robotstxt.org/robotstxt.html)

It is worth noting that [Invidious allows search engines to roam freely](https://invidio.us/robots.txt) over its content, but [Nitter does not](https://nitter.net/robots.txt), choosing to only enable the Twitter link preview service.

**Comment by TheFrenchGhosty** [comment]
In my opinion, the robots.txt should allow everything by default, with an option to disable everything for instance that don't want to be indexed.

The reason is simple:

When people are looking for a Instagram user or post in a search engine they might find that the content is also on a Bibliogram instance, and therefore become discover that they don't need Instagram and that they can look for privacy alternative for the stuff they use without "loosing" what they use.

In short, it spread awareness that alternative-private way exist to watch/read content hosted on privacy-hostile platform.

**Comment by Perflyst** [comment]
As long as I can set my disallow / I am happy with any solution.
The actual nasty bots will crawl your instance anyway (which puts load on it) therefore I also vote for allow everything by default (but exclude search of course if it exists)

**Comment by cloudrac3r** [comment]
A robots.txt file was added in fb6081a7197cf5c4a4181bf94caaa1fb1d1d2669 ([peek live file](https://bibliogram.art/robots.txt)). This should prevent search engines from requesting a high numbered page and causing hundreds or thousands of requests to Instagram behind the scenes. Posts and profile page 1 are still accessible.

A disallow for search will be added to this file when the search feature is added.

Leave a comment if you disagree with these defaults.
